### PR TITLE
[ADR] Api - State providers structure

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | api-platform-3
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | #16567
| License         | MIT

### CODE REVIEW

File: adr/2024_07_15_api_state_providers_structure.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+# API - State Providers Structure
+
+* Status: Accepted
+* Date: 2024-07-15
+
+In effect from version: 2.0
+
+Related PRs: #16543
```
Comment (Rafikooo) [inline adr/2024_07_15_api_state_providers_structure.md]: I'm not certain if this is the correct PR referred to.

Comment (GSadee) [inline adr/2024_07_15_api_state_providers_structure.md]:
```suggestion
Related PRs: #16567
```

File: adr/2024_07_15_api_state_providers_structure.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+Until version 2.0, we supported API Platform version 2, which utilized
+`DataProviders` to deliver data to the API. These `DataProviders` were all
+located within the `Sylius\Bundle\ApiBundle\DataProvider` directory.
+This structure lacked differentiation between the admin and shop sections
+and did not provide clear organization for individual resources. 
+As a result, it became challenging to manage and maintain the codebase
+as the number of providers grew.
+
+With Sylius 2.0, the API Platform has been upgraded to version 3.0.
```
Comment (GSadee) [inline adr/2024_07_15_api_state_providers_structure.md]:
```suggestion
With Sylius 2.0, the API Platform has been upgraded to version 3.
```
or
```suggestion
With Sylius 2.0, the API Platform has been upgraded to version 3.x.
```

File: adr/2024_07_15_api_state_providers_structure.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    * Clear separation between admin and shop providers
+    * Clear separation between resources
+* **Bad, because:**
+    * Need to refactor all providers
+
+## Decision Outcome
+
+Chosen option: `Separate StateProviders by sections and resources`,
+because it provides a clear and manageable way to customize form types
+for different contexts.
```
Comment (GSadee) [inline adr/2024_07_15_api_state_providers_structure.md]: To be honest, I do not understand the link between providers and form types here 🤔

Comment (Wojdylak) [reply adr/2024_07_15_api_state_providers_structure.md]: My mistake :face_with_open_eyes_and_hand_over_mouth:

**Review by GSadee — APPROVED** [review]

**Pull request merged**